### PR TITLE
fix `'Object.assign' should be overriden` (close #1208)

### LINKED ISSUE #1208: 'Object.assign' should be overriden
```js
Error: Cannot call method 'replace' of undefined
at Function.t._error (http://localhost:1351/hammerhead.js:2:4245)
at value (http://localhost:1351/hammerhead.js:2:4719)
at b (http://localhost:1351/r1TwR_TNW!s!utf-8/http://config1.veinteractive.com/scripts/5.0/capture-apps-5.0.0.js:7:15233)
at t._onWindowMessage (http://localhost:1351/hammerhead.js:4:7689)
at r (http://localhost:1351/hammerhead.js:8:26811)
at o (http://localhost:1351/hammerhead.js:4:8738)
at http://localhost:1351/hammerhead.js:3:31718
```

Sites:
http://lingualeo.ru/
http://www.samsung.com/uk/home

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev @churkin

### CODE REVIEW

File: test/client/fixtures/sandbox/node/methods-test.js [hunk 1]
```diff
@@ -367,6 +367,18 @@ if (window.DOMParser && !browserUtils.isIE9) {
...
+if (Object.assign) {
+    test('Object.assign (GH-1208)', function () {
+        var iframe = document.createElement('iframe');
+
+        Object.assign(iframe, { src: '/iframe' });
+
+        strictEqual(iframe.src, urlUtils.getProxyUrl('/iframe', {
+            resourceType: urlUtils.stringifyResourceType({ isIframe: true })
+        }));
+    });
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/methods-test.js]: Add more testcases:
- check returned value of the `Object.assign` 
- all cases from https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
- target is null, primitive type, function

File: src/client/sandbox/node/window.js [hunk 2]
```diff
@@ -86,6 +87,34 @@ export default class WindowSandbox extends SandboxBase {
...
         };
 
+        if (nativeMethods.objectAssign) {
+            window.Object.assign = function (target, ...sources) {
+                const args = [];
+
+                args.push(target);
+
+                if (target && typeof target === 'object' && sources.length) {
+                    for (const source of sources) {
```
Comment (miherlosev) [inline src/client/sandbox/node/window.js]: use the `Array.filter` method

File: src/client/sandbox/node/window.js [hunk 3]
```diff
@@ -86,6 +87,34 @@ export default class WindowSandbox extends SandboxBase {
...
+                        }
+
+                        const sourceKeys = nativeMethods.objectKeys.call(window.Object, source);
+
+                        for (const key of sourceKeys)
+                            window[INSTRUCTION.setProperty](target, key, source[key]);
+                    }
+                }
+                else {
+                    for (const source of sources)
```
Comment (miherlosev) [inline src/client/sandbox/node/window.js]: use the `Array.concat` method

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: test/client/fixtures/sandbox/node/methods-test.js [hunk 4]
```diff
@@ -367,6 +367,96 @@ if (window.DOMParser && !browserUtils.isIE9) {
     });
 }
 
+if (Object.assign) {
+    test('Object.assign', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/methods-test.js]: Split to several tests

Comment (LavrovArtem) [inline test/client/fixtures/sandbox/node/methods-test.js]: right

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
Rebase it

**Comment by miherlosev** [comment]
@churkin FPR

File: src/client/sandbox/node/window.js [hunk 5]
```diff
@@ -86,6 +91,34 @@ export default class WindowSandbox extends SandboxBase {
...
 
+        if (nativeMethods.objectAssign) {
+            window.Object.assign = function (target, ...sources) {
+                let args = [];
+
+                args.push(target);
+
+                const targetType = typeof target;
+
+                if (target && (targetType === 'object' || targetType === 'function') && sources.length) {
```
Comment (LavrovArtem) [inline src/client/sandbox/node/window.js]: ```typeof null === 'object'```

Comment (churkin) [inline src/client/sandbox/node/window.js]: Oh, you are right

**Comment by LavrovArtem** [comment]
ping @churkin

**Pull request merged**